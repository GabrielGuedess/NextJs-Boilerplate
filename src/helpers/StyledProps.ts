export type StyledProps<T> = Required<{
  [K in keyof T as `$${string & K}`]: T[K];
}>;
