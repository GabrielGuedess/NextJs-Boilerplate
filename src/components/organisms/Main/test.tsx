import { renderWithTheme } from 'utils/tests/helpers';

import { Main } from '.';

describe('<Main />', () => {
  it('should render the main correctly', () => {
    const { container } = renderWithTheme(<Main />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
