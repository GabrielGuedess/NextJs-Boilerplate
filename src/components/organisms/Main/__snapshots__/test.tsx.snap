// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the main correctly 1`] = `
.c1 {
  position: relative;
}

.c1::after {
  content: '';
  position: absolute;
  top: 115%;
  left: 0;
  -webkit-transform: scale(1,0.8);
  -ms-transform: scale(1,0.8);
  transform: scale(1,0.8);
  display: block;
  width: 18rem;
  height: calc(18rem / 5);
  background-image: radial-gradient( closest-side, rgba(153,248,255,0.2), rgba(153,248,255,0) );
  border-radius: 100%;
  -webkit-animation: 8s fDuSjn infinite cubic-bezier(1,0.25,0,0.75);
  animation: 8s fDuSjn infinite cubic-bezier(1,0.25,0,0.75);
}

.c4 {
  position: absolute;
  top: 50%;
  width: 100%;
  height: calc(18rem / 2.5);
  margin-top: calc((18rem / 2.5) / -2) !important;
  border-top: solid rgba(0,216,255,0.5) 1px;
  border-right: solid rgba(0,216,255,0.35) 2px;
  border-bottom: solid rgba(0,216,255,0.2) 4px;
  border-left: solid rgba(0,216,255,0) 2px;
  border-radius: 100%;
  -webkit-animation: 1.2s gOiwFb infinite linear;
  animation: 1.2s gOiwFb infinite linear;
}

.c4::after {
  content: '';
  position: absolute;
  top: 50%;
  left: calc((18rem / 25) / -1);
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  display: block;
  width: calc(18rem / 25);
  height: calc(18rem / 25);
  margin-top: calc((18rem / 25) / -2) !important;
  background: #99f8ff;
  border-radius: 50%;
  -webkit-animation: calc(1.2s * 2) dQPgac infinite ease-in-out;
  animation: calc(1.2s * 2) dQPgac infinite ease-in-out;
}

.c2 {
  position: relative;
  width: 18rem;
  height: 18rem;
  -webkit-animation: 8s eWqkYC infinite cubic-bezier(1,0.25,0,0.75);
  animation: 8s eWqkYC infinite cubic-bezier(1,0.25,0,0.75);
}

.c2::before {
  content: '';
  position: absolute;
  top: 50%;
  left: 50%;
  display: block;
  width: calc(18rem / 5);
  height: calc(18rem / 5);
  margin-top: calc((18rem / 5) / -2) !important;
  margin-left: calc((18rem / 5) / -2);
  background: #99f8ff;
  border-radius: 100%;
  box-shadow: 0 0 3px 0 #00d8ff;
  -webkit-animation: 2s kxVzFA infinite cubic-bezier(0.65,0,0.35,1);
  animation: 2s kxVzFA infinite cubic-bezier(0.65,0,0.35,1);
}

.c2 .c3:nth-child(2) {
  -webkit-transform: rotate(60deg);
  -ms-transform: rotate(60deg);
  transform: rotate(60deg);
  -webkit-animation: 1s gOiwFb infinite linear;
  animation: 1s gOiwFb infinite linear;
}

.c2 .c3:nth-child(2)::after {
  -webkit-animation: calc(1s * 2) dQPgac infinite ease-in-out;
  animation: calc(1s * 2) dQPgac infinite ease-in-out;
}

.c2 .c3:nth-child(3) {
  -webkit-transform: rotate(-60deg);
  -ms-transform: rotate(-60deg);
  transform: rotate(-60deg);
  -webkit-animation: 0.8s gOiwFb infinite linear;
  animation: 0.8s gOiwFb infinite linear;
}

.c2 .c3:nth-child(3)::after {
  -webkit-animation: calc(0.8s * 2) dQPgac infinite ease-in-out;
  animation: calc(0.8s * 2) dQPgac infinite ease-in-out;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100vh;
  padding: 3rem;
}

.c5 {
  margin-top: 8rem;
}

<main
  class="c0"
  data-cy="home"
>
  <div
    class="c1"
    data-testid="Logo React"
  >
    <div
      class="c2"
    >
      <div
        class="c3 c4"
      />
      <div
        class="c3 c4"
      />
      <div
        class="c3 c4"
      />
    </div>
  </div>
  <img
    alt="Next.js Logo"
    class="c5"
    data-nimg="1"
    decoding="async"
    fetchpriority="high"
    height="37"
    src="/img/next.svg"
    style="color: transparent;"
    width="180"
  />
</main>
`;
